GNSSLogger/app/src/main/java/com/google/android/apps/location/gps/gnsslogger/FileLogger.java

```diff
@@ -339,6 +366,25 @@ public void onNmeaReceived(long timestamp, String s) {
     }
   }
 
+  @Override
+  public void onSensorChanged(SensorEvent event) {
+    synchronized (mFileLock) {
+      if(mFileWriter == null) {
+        return;
+      }
+      mStringBuilder.setLength(0);
+      mStringBuilder.append(event.sensor.getStringType().substring(event.sensor.getStringType().lastIndexOf('.') + 1)).append(RECORD_DELIMITER);
+      mStringBuilder.append(event.timestamp).append(RECORD_DELIMITER);
+      mStringBuilder.append(java.util.Arrays.toString(event.values).replaceAll("[] ]", "").replace("[", ""));
```

Comment:
This is some slick code golf, but it's a bit hard to follow. I recommend replacing it with something simpler to read. Something like:

```
String.format(
            "Accel,%s,%s,%s,%s,%s,%s,%s",
            event.timestamp,
            event.values[IMU_X_INDEX], 
            event.values[IMU_Y_INDEX],
            event.values[IMU_Z_INDEX], 
            event.values[IMU_X_BIAS_INDEX], 
            event.values[IMU_Y_BIAS_INDEX], 
            event.values[IMU_Z_BIAS_INDEX]);
```
You can extract that into a helper method, and for each sensor, just pass a different sensor type string. The rest seems common.